File: android/src/main/java/com/hemangkumar/capacitorgooglemaps/MapPreferencesControls.java
package com.hemangkumar.capacitorgooglemaps;

import java.util.HashMap;

public class MapPreferencesControls extends JSObjectDefaults {
    public static final String COMPASS_BUTTON_KEY = "isCompassButtonEnabled";
    public static final String INDOOR_LEVEL_PICKER_KEY = "isIndoorLevelPickerEnabled";
    public static final String MAP_TOOLBAR_KEY = "isMapToolbarEnabled";
    public static final String MY_LOCATION_BUTTON_KEY = "isMyLocationButtonEnabled";
    public static final String ZOOM_BUTTONS_KEY = "isZoomButtonsEnabled";

    public MapPreferencesControls() {
        super(new HashMap<String, Object>(){{
            put(COMPASS_BUTTON_KEY, Boolean.TRUE);
            put(INDOOR_LEVEL_PICKER_KEY, Boolean.FALSE);
            put(MAP_TOOLBAR_KEY, Boolean.FALSE);
            put(MY_LOCATION_BUTTON_KEY, Boolean.TRUE);
            put(ZOOM_BUTTONS_KEY, Boolean.FALSE);
        }});
    }
}

File: ios/Plugin/MapPreferencesControls.swift
import Capacitor
import GoogleMaps

class MapPreferencesControls {
    public static let COMPASS_BUTTON_KEY: String! = "isCompassButtonEnabled"
    public static let INDOOR_LEVEL_PICKER_KEY: String! = "isIndoorLevelPickerEnabled"
    public static let MAP_TOOLBAR_KEY: String! = "isMapToolbarEnabled" // (android only)
    public static let MY_LOCATION_BUTTON_KEY: String! = "isMyLocationButtonEnabled"
    public static let ZOOM_BUTTONS_KEY: String! = "isZoomButtonsEnabled" // (android only)

    var _isCompassButtonEnabled: Bool = true
    var isCompassButtonEnabled: Bool! {
        get {
            return _isCompassButtonEnabled
        }
        set (newVal) {
            _isCompassButtonEnabled = newVal ?? true
        }
    }

    var _isIndoorLevelPickerEnabled: Bool = false
    var isIndoorLevelPickerEnabled: Bool! {
        get {
            return _isIndoorLevelPickerEnabled
        }
        set (newVal) {
            _isIndoorLevelPickerEnabled = newVal ?? false
        }
    }

    var _isMyLocationButtonEnabled: Bool = true
    var isMyLocationButtonEnabled: Bool! {
        get {
            return _isMyLocationButtonEnabled
        }
        set (newVal) {
            _isMyLocationButtonEnabled = newVal ?? true
        }
    }

    func updateFromJSObject(object: JSObject) {
        self.isCompassButtonEnabled = object[MapPreferencesControls.COMPASS_BUTTON_KEY] as? Bool;
        self.isIndoorLevelPickerEnabled = object[MapPreferencesControls.INDOOR_LEVEL_PICKER_KEY] as? Bool;
        self.isMyLocationButtonEnabled = object[MapPreferencesControls.MY_LOCATION_BUTTON_KEY] as? Bool;
    }
    
    func getJSObject(_ mapView: GMSMapView) -> JSObject {
        return [
            MapPreferencesControls.COMPASS_BUTTON_KEY: mapView.settings.compassButton,
            MapPreferencesControls.INDOOR_LEVEL_PICKER_KEY: mapView.settings.indoorPicker,
            MapPreferencesControls.MAP_TOOLBAR_KEY: false, // (android only)
            MapPreferencesControls.MY_LOCATION_BUTTON_KEY: mapView.settings.myLocationButton,
            MapPreferencesControls.ZOOM_BUTTONS_KEY: false, // (android only)
        ]
    }
}
